lib/chromic_pdf/assigns_plug.ex

```diff
@@ -0,0 +1,145 @@
+if Code.ensure_loaded?(Plug) and Code.ensure_loaded?(Plug.Crypto) do
+  defmodule ChromicPDF.AssignsPlug do
+    @moduledoc """
+    This module implements an "assigns passing" mechanism between a `print_to_pdf/2` caller process
+    and an internal endpoint.
+
+    ## Usage
+
+    On the caller side:
+
+        ChromicPDF.print_to_pdf({:url, "http://localhost:4000/makepdf"}, assigns: %{hello: :world})
+
+    In your endpoint:
+
+        plug ChromicPDF.AssignsPlug
+
+    The plug makes the assigns available in `conn.assigns` to be used in the HTML template.
+    """
+
+    @behaviour Plug
+
+    import Plug.Conn, only: [assign: 3]
+    alias Plug.{Conn, Crypto}
+
+    @type url :: binary
+    @type assigns :: map
+
+    @default_max_age 3600
+
+    # Default secret_key_base is generated at compile-time which ties the running Chrome instance
+    # to the compiled module.
+    # Potentially this loses requests at the edges when using _external_ chrome instances (i.e.
+    # accessed via TCP) in a clustered environment. Unlikely to be a problem, I think.
+    @default_secret_key_base 32
+                             |> :crypto.strong_rand_bytes()
+                             |> Base.encode64(padding: false)
+                             |> binary_part(0, 32)
```

Comment:
yeah, question for me would rather be, do we need to make it configurable at all? Providing a default and making it configurable seems odd for a secret key :)

Reasoning for having it in code at all: For Phoenix' secret key you want it to stay the same across builds so that sessions that have been signed with build X can still be verified by build Y. However, since this secures only temporary "sessions" between the app and the locally spawned Chrome instance, there should not be any sessions going across builds. Unless you do hot deploys or so.

Wdyt?